### PR TITLE
[Misc] Add unit test for decode_attention Triton kernels

### PR DESCRIPTION
**Pull request by pmanczak**
## Purpose

Adds a device-agnostic unit test for the three decode-attention Triton kernels
in `vllm/v1/attention/ops/triton_decode_attention.py` (extends XPU coverage):

- `_fwd_kernel_stage1` — MHA, via `decode_attention_fwd_normal`
- `_fwd_grouped_kernel_stage1` — GQA, via `decode_attention_fwd_grouped`
- `_fwd_kernel_stage2` — softmax/reduce-V, both paths

## Test Result

 All 16 tests passed.

### CODE REVIEW

**Comment by oonyshch** [comment]
1. `tests/kernels/attention/test_triton_decode_attention.py` tests the same three kernels, so as this PR expands that coverage have you considered placing it in the same file? per .md guidance in repo new files should only be created when no nearby-suite fits
2. pls run `ruff format`

File: tests/kernels/attention/test_decode_attention_torch_ref.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    H_q, H_kv = 8, 2
+    (q, k_buffer, v_buffer, o, lse, req_to_tokens, b_seqlen, att_out,
+     k_scale, v_scale, sm_scale) = _make_inputs(
+        B, H_q, H_kv, D_head, seq_len, num_kv_splits, PAGE_SIZE, shuffle_pages)
+
+    decode_attention_fwd_grouped(
+        q, k_buffer, v_buffer, o, lse,
+        req_to_tokens, b_seqlen, att_out,
+        num_kv_splits, sm_scale, PAGE_SIZE,
+        logit_cap=0.0, k_scale=k_scale, v_scale=v_scale,
```
Comment (oonyshch) [inline tests/kernels/attention/test_decode_attention_torch_ref.py]: maybe parametrize logit_cap over [0.0, 30.0] and apply the same tanh cap in torch_decode_attention before logsumexp/softmax. its passed as 0.0 always

File: tests/kernels/attention/test_decode_attention_torch_ref.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  * _fwd_kernel_stage2          the softmax/reduce-V stage used by both
+
+The pure-PyTorch implementation is the oracle, so the test is device-agnostic
+and runs wherever current_platform points (CUDA, ROCm or XPU).
+
+Note: decode_attention_fwd_normal / decode_attention_fwd_grouped do not
+default k_scale / v_scale, and the kernels call tl.load(k_scale) /
+tl.load(v_scale) unconditionally, so both must be passed as torch.Tensor
+pointers (not Python floats / None).
+"""
```
Comment (oonyshch) [inline tests/kernels/attention/test_decode_attention_torch_ref.py]: maybe call the public `decode_attention_fwd` instead of the private `_normal`/`_grouped` helpers ; it dispatches by `kv_group_num` and defaults the scales, dropping the tensor-pointer workaround below and this note

File: tests/kernels/attention/test_decode_attention_torch_ref.py (hunk 1 above)
Comment (pmanczak) [reply tests/kernels/attention/test_decode_attention_torch_ref.py]: Addressed. logit_cap is parametrized and the tanh cap is applied in the oracle before logsumexp/softmax. I used 1.0 not 30.0 because with sm_scale=1/√D scores are ~unit-variance, so 30*tanh(x/30)≈x. 1.0 actually exercises the cap, verified by mutation test.
